src/rpc/stats.cpp

```diff
@@ -0,0 +1,124 @@
+#include <rpc/stats.h>
+
+#include <rpc/server.h>
+#include <rpc/util.h>
+
+bool CRPCStats::add(const std::string& name, const int64_t latency, const int64_t payload)
+{
+    UniValue data(UniValue::VOBJ),
+            latencyObj(UniValue::VOBJ),
+            payloadObj(UniValue::VOBJ),
+            history(UniValue::VARR);
+
+    int64_t min_latency = latency,
+            avg_latency = latency,
+            max_latency = latency,
+            min_payload = payload,
+            avg_payload = payload,
+            max_payload = payload,
+            count       = 1;
+
+    auto stats = statsRPC.get(name);
+    if (!stats.empty()) {
+        count = stats["count"].get_int() + 1;
+
+        latencyObj = stats["latency"].get_obj();
+        if (latency < latencyObj["min"].get_int()) min_latency = latency;
+        if (latency > latencyObj["max"].get_int()) max_latency = latency;;
+        avg_latency = latencyObj["avg"].get_int() + (latency - latencyObj["avg"].get_int()) / count;
```

Comment:
Why not `avg_latency = (latencyObj["avg"].get_int64() + latency) / count` ?

Reply:
Would not work.
Let's take latency values 2 3 10:
first avg: 2 / 1 == 2
second: (2 + 3) / 2 == 2.5 
third: (2.5 + 10) / 3 == 4.16~ (should be 5)

See https://math.stackexchange.com/questions/22348/how-to-add-and-subtract-values-from-an-average for formula explanation

Reply:
Alright then can we do just, (old value + new value) / 2, i still think that formula is bit complicated than it should. If it still needed i'm fine.

Reply:
would not work either
taking third from above :
(2.5 + 10) / 2 == 7.5

What could work is 
(old value * (count - 1) + new value) / count 
(2.5 * 2 + 10) / 3 == 5

Reply:
Ok

Reply:
Running avg. looks good. Don't store the cache as JSON (Univalue), however. Just use raw objects. It's pretty pointless to do this per call. Just serialize the object on output.